public/app/core/components/NativeScrollbar.tsx

```diff
@@ -1,28 +1,33 @@
 import { css, cx } from '@emotion/css';
 import { useEffect, useRef } from 'react';
 
+import { config } from '@grafana/runtime';
 import { CustomScrollbar, useStyles2 } from '@grafana/ui';
 
 type Props = Parameters<typeof CustomScrollbar>[0];
 
 // Shim to provide API-compatibility for Page's scroll-related props
+// when bodyScrolling is enabled, this is a no-op
+// TODO remove this shim completely when bodyScrolling is enabled
```

Comment:
Unfortunately, I think we should maintain compatibility here, and have these properties return a ref(?) to the body, and scroll the body.

Ideally, we shouldn't have any changes to the DashboardPage for the scroll restoration.

Reply:
this is gonna sound sassier than i'm intending, but... why? 🤔

the only place that uses it is `DashboardPage` and it's not exposed to plugins via `PluginPage`. feels like this is a good time to decouple it from the `Page` interface. unless we're thinking `Page` is the correct place for this logic to live 🤔